java-checks/src/main/java/org/sonar/java/checks/design/AbstractCouplingChecker.java

```diff
@@ -50,48 +42,20 @@
 import org.sonar.plugins.java.api.tree.VariableTree;
 import org.sonar.plugins.java.api.tree.WildcardTree;
 
-@Rule(key = "S1200")
-public class ClassCouplingCheck extends BaseTreeVisitor implements JavaFileScanner {
+public abstract class AbstractCouplingChecker extends BaseTreeVisitor implements JavaFileScanner {
 
-  private static final int DEFAULT_MAX = 20;
+  protected final Deque<Set<String>> nesting = new LinkedList<>();
+  protected Set<String> types;
+  protected JavaFileScannerContext context;
 
-  @RuleProperty(
-    key = "max",
-    description = "Maximum number of classes a single class is allowed to depend upon",
-    defaultValue = "" + DEFAULT_MAX)
-  public int max = DEFAULT_MAX;
-
-  private final Deque<Set<String>> nesting = new LinkedList<>();
-  private Set<String> types;
-  private JavaFileScannerContext context;
+  abstract void checkTypes(@Nullable Tree type);
```

Comment:
Because checkTypes is invoked in the `visitVariable` method, we should document what is the expected behavior for an implementation of the method

Reply:
Should the return type be changed to `Set<String>` and return the new types added to the field?